Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,640 @@
+                                     const NSDecimal* leftOperand,
+                                     const NSDecimal* rightOperand,
+                                     NSRoundingMode roundingMode) {
+    THROW_NS_IF_FALSE(E_INVALIDARG, (leftOperand != NULL) && (rightOperand != NULL) && (result != NULL));
+    VALIDATE_ARITHMETIC_INPUT(result, leftOperand, rightOperand);
+    // Convert the subtraction into an addition.
+    // L - R -> L + (-R)
+    // flip the sign, if and only if it's non zero and a valid number
+    NSDecimal right;
+    NSDecimalCopy(&right, rightOperand);
+    right._isNegative = (IS_ZERO(&right) || NSDecimalIsNotANumber(&right)) ? right._isNegative : !right._isNegative;
+
+    return NSDecimalAdd(result, leftOperand, &right, roundingMode);
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalMultiply(NSDecimal* result,
+                                     const NSDecimal* leftOperand,
+                                     const NSDecimal* rightOperand,
+                                     NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalDivide(NSDecimal* result,
+                                   const NSDecimal* leftOperand,
+                                   const NSDecimal* rightOperand,
+                                   NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSString* NSDecimalString(const NSDecimal* dcm, id locale) STUB_METHOD {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalPower(NSDecimal* result, const NSDecimal* number, NSUInteger power, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+void NSDecimalRound(NSDecimal* result, const NSDecimal* number, NSInteger scale, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+}
+
+static void _InternalDecimalCopy(NSDecimal* destination, NSDecimalInternal* source) {
+    if (!source || !destination) {
+        return;
+    }
+    NSDecimalCopy(destination, &(source->decimal));
+    memcpy(destination->_mantissa, source->_mantissa, NSDecimalMaxSize * sizeof(unsigned short));
+}
+
+static NSComparisonResult _compareMantissa(NSDecimal* left, NSDecimal* right) {
+    // Note: All inputs have been normalized via NSDecimalNormalize
+    if (left->_length > right->_length) {
+        return NSOrderedDescending;
+    }
+
+    if (left->_length < right->_length) {
+        return NSOrderedAscending;
+    }
+
+    // same length
+    for (int index = right->_length - 1; index >= 0; index--) {
```

Comment:
looks likes it.

---
In reply to: [81231952](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r81231952) [](ancestors = 81231952)